model/task/task.go

```diff
@@ -1273,6 +1260,35 @@ func SetTasksScheduledTime(tasks []Task, scheduledTime time.Time) error {
 	return nil
 }
 
+// GetTaskIdBetweenIds gets the task between two task given that they are
+// from the same project, requester, build variant, and display name. The
+// order of the ID's does not matter and if the task passed cannot have a
+// middle (i.e. it is sequential tasks or the same task) it will return the
+// the first task given.
+func GetMidwayTask(t1, t2 Task) (*Task, error) {
+	// The tasks should be the same build variant, display name, project, and requester.
+	if t1.BuildVariant != t2.BuildVariant {
+		return nil, errors.Errorf("given tasks have differing build variants '%s' and '%s'", t1.BuildVariant, t2.BuildVariant)
+	}
+	if t1.DisplayName != t2.DisplayName {
+		return nil, errors.Errorf("given tasks have differing display names '%s' and '%s'", t1.DisplayName, t2.DisplayName)
+	}
+	if t1.Project != t2.Project {
+		return nil, errors.Errorf("given tasks have differing projects '%s' and '%s'", t1.Project, t2.Project)
+	}
+	if t1.Requester != t2.Requester {
+		return nil, errors.Errorf("given tasks have differing requesters '%s' and '%s'", t1.Requester, t2.Requester)
```

Comment:
I think it would be worth testing the validation logic at least a little bit, since we're now requiring that they're the same task definition.